src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp

```diff
@@ -98,8 +98,17 @@ void TargetedMovementGeneratorMedium<T,D>::_setTargetLocation(T* owner, bool ini
         if ((!initial || (owner->movespline->Finalized() && this->GetMovementGeneratorType() == CHASE_MOTION_TYPE)) && i_target->IsWithinDistInMap(owner, dist) && i_target->IsWithinLOS(owner->GetPositionX(), owner->GetPositionY(), owner->GetPositionZ()))
             return;
 
-        // Xinef: Fix follow angle for hostile units
         float angle = i_angle;
+
+        if (i_target->GetTypeId() == TYPEID_PLAYER && owner->ToCreature()->GetCreatureType() == CREATURE_TYPE_NON_COMBAT_PET)
```

Comment:
Is it somewhere checked that `owner->ToCreature()` is safe? Is the type checked somewhere?  I cannot see it.

Reply:
@Stoabrogga

Reply:
Yes, you're right, it is better to check if the owner is a creature before determining the creature type:
```
if (i_target->GetTypeId() == TYPEID_PLAYER)
{
	Creature* creature = owner->ToCreature();

	if (creature && creature->GetCreatureType() == CREATURE_TYPE_NON_COMBAT_PET)
	{
		// fix distance and angle for vanity pets
		dist = 0.3f;
		angle = PET_FOLLOW_ANGLE + M_PI * 0.2f;
		size = i_target->GetCombatReach() - i_target->GetObjectSize();
	}
}
```
Can someone commit this directly or should I create another PR for this?

Reply:
please open a PR, we should avoid committing directly